### PR TITLE
feature: add xgboost framework version 1.2-1

### PR DESCRIPTION
**Pull request by edwardjkim**
*Description of changes:*
This PR adds Sagemaker XGBoost version `1.2-1`. In `1.2-1` we drop the suffix `-cpu-py3` since the image supports both CPU and GPU.

*Testing done:*
`tox tests/unit`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by edwardjkim** [comment]
@iyerr3 Can you review the PR? Specifically the error message here: https://github.com/aws/sagemaker-python-sdk/pull/1899/files#diff-42337fa10ff460fd141c4f009a88db73R17

File: src/sagemaker/image_uri_config/xgboost.json [hunk 1]
```diff
@@ -125,6 +125,35 @@
                 "us-west-2": "246618743249"
             },
             "repository": "sagemaker-xgboost"
+        },
+        "1.2-1": {
```
Comment (ChuyangDeng) [inline src/sagemaker/image_uri_config/xgboost.json:129]: Did you also update the `version_aliases` on this file?

**Review by ChuyangDeng — CHANGES_REQUESTED** [review]

File: src/sagemaker/image_uri_config/xgboost.json (hunk 1 above)
Comment (iyerr3) [reply src/sagemaker/image_uri_config/xgboost.json:129]: @edwardjkim We should add `1.2` as an alias. 
@ChuyangDeng we can hold off on the other alias PR if there are conflicts with this PR.

Comment (edwardjkim) [reply src/sagemaker/image_uri_config/xgboost.json:129]: @iyerr3 Are we sure that we want to add `1.2` as an alias? Doesn't that kind of defeat the purpose of having the container version `-1` in `1.2-1`?

**Comment by edwardjkim** [comment]
> ### AWS CodeBuild CI Report
> * CodeBuild project: sagemaker-python-sdk-pr
> * Commit ID: [f2c2507](https://github.com/aws/sagemaker-python-sdk/commit/f2c25072b4ad28eb3a5cb1f8f4af1c3a56660e6e)
> * Result: FAILED
> * [Build Logs](https://wsr3865w1k.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=0e2c5a82-647c-4679-a75d-907b2f465e0c%2Fbuild.log) (available for 30 days)
> 
> _Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)_

@ChuyangDeng I'm confused. Why is `'repository'` getting passed to `xgboost_latest_py_version`?

File: src/sagemaker/image_uri_config/xgboost.json (hunk 1 above)
Comment (edwardjkim) [reply src/sagemaker/image_uri_config/xgboost.json:129]: @ChuyangDeng I talked to Rahul offline. We are going to hold off on `version_aliases` and revisit later.

**Review by ChuyangDeng — APPROVED** [review]

**Pull request merged**